pkg/operator/ceph/cluster/cluster_external.go

```diff
@@ -181,3 +197,104 @@ func validateExternalClusterSpec(cluster *cluster) error {
 
 	return nil
 }
+
+func createExternalMetricsEndpoints(cluster *cluster) *v1.Endpoints {
```

Comment:
Seems like most of this new code should be in the `mgr` package instead of this `cluster` package.